### DISCUSSION TITLE
x-for with $parent

### DISCUSSION THREAD

**Discussion by leo-petrucci** [alpine-collective/alpine-magic-helpers#126] [Q&A] [answered]
I've been trying to get this working for a while but it just doesn't seem to be working.

```
<div x-data="{ comments: ['test', 'test2'] }">
  <div class="prose" id="commentcarp__comments" x-data>
    <p x-text="$parent.comments[0]"></p>
    <template x-for="item in $parent.comments" :key="item">
      <p x-text="item"></p>
    </template>
  </div>
</div>
```

I need to access the `$parent`'s comments array but nothing is being rendered.

I can tell the array exists because I can see the first item with `<p x-text="$parent.comments[0]"></p>` but as soon as I try iterating through it nothing gets rendered.

I made a [codepen of the issue](https://codepen.io/creativiii-the-scripter/pen/abpgXma?editors=1000)

What am I missing?

**Comment by SimoTod** [comment, marked as answer]
The codepen looks okay, there's probably a bug to fix, we'll look into it as soon as we can. In the meantime, you can use this workaround using indexes: https://codepen.io/SimoTod/pen/abpgMLO

**Reply by leo-petrucci** [reply]
Glad to know I'm not crazy! Thank you for the workaround 😄
